rlclientlib/live_model_impl.cc

```diff
@@ -518,7 +518,7 @@ int live_model_impl::init_loggers(api_status* status)
 
   // TODO: Use a specific episode message type (for now it is the same with the observation logger, using
   // observation_logger_facade).
-  if (_configuration.get(name::EPISODE_EH_HOST, nullptr) != nullptr)
+  if (_configuration.get(name::EPISODE_SENDER_IMPLEMENTATION, nullptr) != nullptr)
```

Comment:
Which conditions should we be checking exactly? I know only checking for the EH host is wrong since it breaks file sending. But if USE_AZURE_FACTORIES is true then this check could also fail

Reply:
@zwd-ms do you know?

Reply:
I think what this should really check is whether the client is sending multistep events. Only multistep needs the episode logger. The if condition here checks if it is multistep, then initialize the episode logger.
So looks like the check should be whether the configuration file has either eventhub or file logger config for episode (meaning it is using multistep), then do the initialization.